### PR TITLE
Add to cli "--debug" flag to print each task outputs

### LINKED ISSUE #415: Add to cli a "--debug" flag, to print each task Outputs  [3]
That way it will be easier to debug issues with flows...
should be added to the run cmd.

With different color to the outputs...

@orius123 @oritstone @Bonczidai @tudorlesan @SamMarkowitz @lirantal WDYT?

### PR DESCRIPTION
**Pull request by maayan-avraham**
Fixes #415
Signed-off-by: Maayan Avraham maayan.avraham@hpe.com

### CODE REVIEW

**Comment by orius123** [comment]
Seems weird to me to have quiet flag and a debug flag, it looks like levels of verbosity? Nope?

**Comment by maayan-avraham** [comment]
@Bonczidai  Please review.

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java [hunk 1]
```diff
@@ -85,7 +89,20 @@ public synchronized void onEvent(ScoreEvent scoreEvent) throws InterruptedExcept
             case ScoreLangConstants.EVENT_OUTPUT_END:
                 Map<String, Serializable> outputs = extractOutputs(data);
```
Comment (Bonczidai) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java]: with this new functionality, I would rewrite this section little bit: test whether the event refers to a task or flow end and assign behavior accordingly (for DRY - now you are calling `extractNotEmptyOutputs` twice for each task). Thay way you would only one `extractOutputs method`. WDYT?

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java [hunk 2]
```diff
@@ -40,12 +40,16 @@
     public static final String SCORE_ERROR_EVENT_MSG = "Score Error Event :";
     public static final String FLOW_FINISHED_WITH_FAILURE_MSG = "Flow finished with failure";
     public static final String EXEC_START_PATH = "0";
-    public static final int OUTPUT_VALUE_LIMIT = 40;
+    public static final int OUTPUT_VALUE_LIMIT = 100;
     private final static String TASK_PATH_PREFIX = "- ";
 
     private AtomicBoolean flowFinished = new AtomicBoolean(false);
     private AtomicReference<String> errorMessage = new AtomicReference<>("");
+    private boolean isDebugMode = false;
```
Comment (Bonczidai) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java]: I prefer names like `debugMode` (matter of taste)

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java [hunk 3]
```diff
@@ -97,29 +114,41 @@ public synchronized void onEvent(ScoreEvent scoreEvent) throws InterruptedExcept
...
+        return new HashMap<>();
+    }
 
+    public static Map<String, Serializable> extractNotEmptyOutputs(Map<String, Serializable> data) {
+        @SuppressWarnings("unchecked") Map<String, Serializable> outputs = (Map<String, Serializable>) data.get(LanguageEventData.OUTPUTS);
+        if (MapUtils.isNotEmpty(outputs)) {
+            Iterator<Map.Entry<String, Serializable>> iterator = outputs.entrySet().iterator();
+            while (iterator.hasNext()) {
+                Map.Entry<String, Serializable> output = iterator.next();
+                if (output.getValue() == null || StringUtils.isEmpty(output.getValue().toString())) {
```
Comment (Bonczidai) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java]: simpler form: `StringUtils.isEmpty(output.getValue())` - does the same thing

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java [hunk 4]
```diff
@@ -97,29 +114,41 @@ public synchronized void onEvent(ScoreEvent scoreEvent) throws InterruptedExcept
...
+                    iterator.remove();
```
Comment (Bonczidai) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java]: do we really need to consume (remove) the output?

File: cloudslang-cli/src/test/java/io/cloudslang/lang/cli/SlangCLITest.java [hunk 5]
```diff
@@ -92,31 +92,65 @@ public void testRunQuietlyValidFilePathSync() throws Exception {
...
+
+        verify(compilerHelperMock).compile(contains(FLOW_PATH_BACKSLASH), isNull(List.class));
+        verify(ScoreServicesMock).trigger(eq(emptyCompilationArtifact), anyMapOf(String.class, Serializable.class), anyMapOf(String.class, Serializable.class));
+
+        Assert.assertEquals("method threw exception", null, cr.getException());
+        Assert.assertEquals("success should be true", true, cr.isSuccess());
+    }
+
+    @Test (timeout = DEFAULT_TIMEOUT)
+    public void testRunDebugValidFilePathSync() throws Exception {
```
Comment (Bonczidai) [inline cloudslang-cli/src/test/java/io/cloudslang/lang/cli/SlangCLITest.java]: I think that the `sync` test is enough (we do not use `debug` variable in `async` case) - rather add a test case when debug mode is off

**Comment by Bonczidai** [comment]
:+1: @maayan-avraham please handle comments

I agree with @orius123 regarding verbosity (maybe we should open an issue) - @oritstone WDYT?

**Comment by oritstone** [comment]
I also agree with @orius123 I think we should merge the two flags for --verbose flag with verbosity levels

File: cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java (hunk 3 above)
Comment (maayan-avraham) [inline cloudslang-cli/src/main/java/io/cloudslang/lang/cli/services/SyncTriggerEventListener.java]: without toString()  - it's an error

File: cloudslang-cli/src/test/java/io/cloudslang/lang/cli/SlangCLITest.java (hunk 5 above)
Comment (maayan-avraham) [inline cloudslang-cli/src/test/java/io/cloudslang/lang/cli/SlangCLITest.java]: agree. test case with debug mode off is already exists.

**Comment by Bonczidai** [comment]
:ok_hand:

**Pull request merged**